src/network/networktool.cpp

```diff
@@ -38,19 +38,38 @@ void NetworkTool::checkNewVersionInGitHubPayload(QJsonDocument& jsonDocument)
 {
     const static QString projectTag = QStringLiteral(GIT_TAG);
 
+    auto semverToInt = [](const QString& version) -> int {
+        int result = 0;
+        QStringList subVersion = version.right(1).split('.');
+
+        // Check if subVersion is valid
+        if(subVersion.length() > 3)
+        {
+            return result;
+        }
+
+        // Transform version to int
+        for(int i = 0; i < subVersion.length(); i++)
+        {
+            result += subVersion[i].toInt()<<(3-i);
+        }
+        return result;
+    };
+
     // Check if version is a normal release
     // For everything else we can presume test release
-    static QRegularExpression releaseTagRegex(QStringLiteral(R"([v]\d+\.\d+)"));
+    // This regex will check for vX, vX.X and vX.X.X
+    static QRegularExpression releaseTagRegex(QStringLiteral(R"([v]\d+(\.\d+){0,})"));
     QRegularExpressionMatch regexMatch = releaseTagRegex.match(projectTag);
     // Check if this actual version is a real release (normal or test)
     // Alarm people to download the correction version !
-    float actualVersion = regexMatch.hasMatch() ? projectTag.right(1).toFloat() : -1.0;
+    int actualVersion = regexMatch.hasMatch() ? semverToInt(projectTag) : -1.0;
```

Comment:
You need to change also https://github.com/bluerobotics/ping-viewer/pull/333/commits/e4acb8c3ec5277f62f7e84bb31e9ce13d27f3399#diff-55288e377d501245cba94fbcda302a67R102